### PR TITLE
refactor: Refactor data scratch-ipblocks

### LINKED ISSUE #1746: Data scratch - ip blocks
# Plan

According to the specification described in epic [#1727](https://github.com/Flank/flank/issues/1727)
add abstraction and implementation for the following part of the data layer:

## Abstraction

`ftl/data/IpBlock.kt`

```kotlin
package ftl.data

val fetchIpBLocks: IpBlock.Fetch get() = TODO()

data class IpBlock(
    val block: String,
    val form: String,
    val addedDate: String
) {

    interface Fetch : () -> List<IpBlock>
}
```

## Target

* `ftl/environment/ListIPBlocks.kt`

## Adapter

`ftl/adapter/GoogleIpBlockFetch.kt`

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1746

- Code is refactored according to the description in #1746
- command `flank ip-blocks list` runs without any problems

## Checklist

- [x] Unit tested

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/adapter/environment/ListIPBlocks.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package ftl.adapter.environment
+
+import com.google.testing.model.Date
+import com.google.testing.model.DeviceIpBlock
+import ftl.data.IpBlock
+import ftl.gc.deviceIPBlocks
+import ftl.reports.api.twoDigitString
+import java.util.Objects
+
+fun ipBlocksList() = deviceIPBlocks().toIpBlocks()
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/adapter/environment/ListIPBlocks.kt:10]:
```suggestion
fun ipBlocksList(): List<IpBlock> = deviceIPBlocks().toIpBlocks()
```

It's good to provide return type explicitly for public functions.
It makes the implementation a little bit easier to understand.

**Review by jan-gogo — CHANGES_REQUESTED** [review]
The files:
* IpBlocksToTable.kt
* ListIPBlocks.kt

are not a part of the adapters layer because they don't have any references to google API libs additionally they are widely used in the domain layer. According to the diagram below, the code from adapters cannot be used directly in the domain layer.

architecture diagram

Please move back those two files to the `ftl.environment` and leave them for further domain refactor.

**Comment by jan-gogo** [comment]
@piotradamczyk5 Sorry for the mistake, I have copy-pasted the wrong file name, should be `TestEnvironmentInfo.kt` instead of `ListIPBlocks.kt`

**Review by jan-gogo — CHANGES_REQUESTED** [review]
@piotradamczyk5 Sorry for the mistake, I have copy-pasted the wrong file name, should be TestEnvironmentInfo.kt instead of ListIPBlocks.kt

**Comment by piotradamczyk5** [comment]
> The files:
> 
> * IpBlocksToTable.kt
> * ~ListIPBlocks.kt~ TestEnvironmentInfo.kt
> 
> are not a part of the adapters layer because they don't have any references to google API libs additionally they are widely used in the domain layer. According to the diagram below, the code from adapters cannot be used directly in the domain layer.
> 
> architecture diagram
> 
> Please move back those two files to the `ftl.environment` and leave them for further domain refactor.

For me data and adapter layers are badly named:
`adapter` layer is `data` while `data` is `adapter`

**Comment by jan-gogo** [comment]
@piotradamczyk5 I do not agree. The adapter is an implementation that is adapting the library code or client implementation to the data structures and interfaces declared in the data layer.

**Comment by piotradamczyk5** [comment]
> @piotradamczyk5 I do not agree. The adapter is an implementation that is adapting the library code or client implementation to the data structures and interfaces declared in the data layer.

Discussed on slack with @jan-gogo . 
The new package structure will be 
`presentation` > `domain` > `api` > `adapter` > `client`

**Review by jan-gogo — APPROVED** [review]

**Review by adamfilipow92 — APPROVED** [review]

**Pull request merged**